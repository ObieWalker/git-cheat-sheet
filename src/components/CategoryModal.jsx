import React from 'react';
import { Modal } from 'react-bootstrap'

const CategoryModal= ({category, addCategory, editCategory, updateDetails, show, onHide, errors}) => {
  return (
    <div>
      <Modal
          show={show}
          onHide={onHide}
          aria-labelledby="contained-modal-title"
          >
          <Modal.Header closeButton>
            <Modal.Title id="contained-modal-title">Category details
            </Modal.Title>
            </Modal.Header>
            <Modal.Body style={{ width: '100%', marginLeft: 'auto', marginRight: 'auto'}}>
            <label htmlFor="name" className="active">
              Category Title:
            </label>
            {errors.categoryTitle && (
              <span className="alert-danger"><br />
                {errors.categoryTitle}
              </span>
            )}
            <input
              type="text"
              className="form-control"
              id="categoryTitle"
              name="categoryTitle"
              defaultValue={category.name || ''}
              required
              onChange={updateDetails}
            />
            <br />
            </Modal.Body>
            <Modal.Footer>
              {
                category._id ? 
                <button
                  type="submit"
                  id="add-grocery"
                  className="waves-effect waves-dark
                  btn right hoverable dark-green btn btn-primary"
                  onClick={() => editCategory(category._id)}
                  >Edit Category
                </button>
              :
                <button
                  type="submit"
                  id="add-grocery"
                  className="waves-effect waves-dark
                  btn right hoverable dark-green btn btn-primary"
                  onClick={addCategory}
                  >Add Category
                </button>
            }

            
            </Modal.Footer>
          </Modal>
    </div>
  );
}

export default CategoryModal;
