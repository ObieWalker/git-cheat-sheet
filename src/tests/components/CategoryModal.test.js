import React from 'react';
import ReactDOM from 'react-dom';
import expect from 'expect';
import { shallow } from 'enzyme';
import CategoryModal from '../../components/CategoryModal';

let mounted;
let props;
let wrapper;
const addCategory = jest.fn(() => Promise.resolve({}));
const editCategory = jest.fn(() => Promise.resolve({}));
const updateDetails = jest.fn(() => Promise.resolve({}));
const show  = false
const onHide = jest.fn(() => Promise.resolve({}));
const errors = {
  categoryTitle : "there is error"
}
const categoryTitle = ''
const category = {

}

const getComponent = () => {
  if (!mounted) {
    props = {
      addCategory, 
      editCategory,
      updateDetails, 
      show, 
      onHide, 
      errors, 
      categoryTitle, 
      category 
    };
    mounted = shallow(<CategoryModal {...props} />);
  }
  return mounted;
};

describe('cheats Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('should fill in form', () => {
    wrapper = getComponent();
    wrapper
    .find('.ca-title')
    .simulate('change', { target: { name: 'categoryTitle', value: 'obi' } });
  });

  it('should click on add category', () => {
    wrapper = getComponent();
    wrapper
    .find('.waves-effect')
    .simulate('click');
  });

  it('should click on add category', () => {
    wrapper = getComponent();
    wrapper.setProps({ category : {
      _id: '1212121'
    }})
    .find('.waves-effect')
    .simulate('click');
  });
});

