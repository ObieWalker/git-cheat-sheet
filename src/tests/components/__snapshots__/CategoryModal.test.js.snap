// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cheats Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoryModal
    addCategory={[MockFunction]}
    category={Object {}}
    categoryTitle=""
    editCategory={[MockFunction]}
    errors={
      Object {
        "categoryTitle": "there is error",
      }
    }
    onHide={[MockFunction]}
    show={false}
    updateDetails={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        animation={true}
        aria-labelledby="contained-modal-title"
        autoFocus={true}
        backdrop={true}
        bsClass="modal"
        dialogComponentClass={[Function]}
        enforceFocus={true}
        keyboard={true}
        manager={
          ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          }
        }
        onHide={[MockFunction]}
        renderBackdrop={[Function]}
        restoreFocus={true}
        show={false}
      >
        <ModalHeader
          bsClass="modal-header"
          closeButton={true}
          closeLabel="Close"
        >
          <ModalTitle
            bsClass="modal-title"
            componentClass="h4"
            id="contained-modal-title"
          >
            Category details
          </ModalTitle>
        </ModalHeader>
        <ModalBody
          bsClass="modal-body"
          componentClass="div"
          style={
            Object {
              "marginLeft": "auto",
              "marginRight": "auto",
              "width": "100%",
            }
          }
        >
          <label
            className="active"
            htmlFor="name"
          >
            Category Title:
          </label>
          <span
            className="alert-danger"
          >
            <br />
            there is error
          </span>
          <input
            className="form-control ca-title"
            defaultValue=""
            id="categoryTitle"
            name="categoryTitle"
            onChange={[MockFunction]}
            required={true}
            type="text"
          />
          <br />
        </ModalBody>
        <ModalFooter
          bsClass="modal-footer"
          componentClass="div"
        >
          <button
            className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
            id="add-grocery"
            onClick={[MockFunction]}
            type="submit"
          >
            Add Category
          </button>
        </ModalFooter>
      </Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animation": true,
        "aria-labelledby": "contained-modal-title",
        "autoFocus": true,
        "backdrop": true,
        "bsClass": "modal",
        "children": Array [
          <ModalHeader
            bsClass="modal-header"
            closeButton={true}
            closeLabel="Close"
          >
            <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
              id="contained-modal-title"
            >
              Category details
            </ModalTitle>
          </ModalHeader>,
          <ModalBody
            bsClass="modal-body"
            componentClass="div"
            style={
              Object {
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "100%",
              }
            }
          >
            <label
              className="active"
              htmlFor="name"
            >
              Category Title:
            </label>
            <span
              className="alert-danger"
            >
              <br />
              there is error
            </span>
            <input
              className="form-control ca-title"
              defaultValue=""
              id="categoryTitle"
              name="categoryTitle"
              onChange={[MockFunction]}
              required={true}
              type="text"
            />
            <br />
          </ModalBody>,
          <ModalFooter
            bsClass="modal-footer"
            componentClass="div"
          >
            <button
              className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
              id="add-grocery"
              onClick={[MockFunction]}
              type="submit"
            >
              Add Category
            </button>
          </ModalFooter>,
        ],
        "dialogComponentClass": [Function],
        "enforceFocus": true,
        "keyboard": true,
        "manager": ModalManager {
          "add": [Function],
          "containers": Array [],
          "data": Array [],
          "handleContainerOverflow": true,
          "hideSiblingNodes": true,
          "isTopModal": [Function],
          "modals": Array [],
          "remove": [Function],
        },
        "onHide": [MockFunction],
        "renderBackdrop": [Function],
        "restoreFocus": true,
        "show": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-header",
            "children": <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
              id="contained-modal-title"
            >
              Category details
            </ModalTitle>,
            "closeButton": true,
            "closeLabel": "Close",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-title",
              "children": "Category details",
              "componentClass": "h4",
              "id": "contained-modal-title",
            },
            "ref": null,
            "rendered": "Category details",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-body",
            "children": Array [
              <label
                className="active"
                htmlFor="name"
              >
                Category Title:
              </label>,
              <span
                className="alert-danger"
              >
                <br />
                there is error
              </span>,
              <input
                className="form-control ca-title"
                defaultValue=""
                id="categoryTitle"
                name="categoryTitle"
                onChange={[MockFunction]}
                required={true}
                type="text"
              />,
              <br />,
            ],
            "componentClass": "div",
            "style": Object {
              "marginLeft": "auto",
              "marginRight": "auto",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Category Title:",
                "className": "active",
                "htmlFor": "name",
              },
              "ref": null,
              "rendered": "Category Title:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <br />,
                  "there is error",
                ],
                "className": "alert-danger",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "there is error",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control ca-title",
                "defaultValue": "",
                "id": "categoryTitle",
                "name": "categoryTitle",
                "onChange": [MockFunction],
                "required": true,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-footer",
            "children": <button
              className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
              id="add-grocery"
              onClick={[MockFunction]}
              type="submit"
            >
              Add Category
            </button>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Category",
              "className": "waves-effect waves-dark btn right hoverable dark-green btn btn-primary",
              "id": "add-grocery",
              "onClick": [MockFunction],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Add Category",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          animation={true}
          aria-labelledby="contained-modal-title"
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[MockFunction]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <ModalHeader
            bsClass="modal-header"
            closeButton={true}
            closeLabel="Close"
          >
            <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
              id="contained-modal-title"
            >
              Category details
            </ModalTitle>
          </ModalHeader>
          <ModalBody
            bsClass="modal-body"
            componentClass="div"
            style={
              Object {
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "100%",
              }
            }
          >
            <label
              className="active"
              htmlFor="name"
            >
              Category Title:
            </label>
            <span
              className="alert-danger"
            >
              <br />
              there is error
            </span>
            <input
              className="form-control ca-title"
              defaultValue=""
              id="categoryTitle"
              name="categoryTitle"
              onChange={[MockFunction]}
              required={true}
              type="text"
            />
            <br />
          </ModalBody>
          <ModalFooter
            bsClass="modal-footer"
            componentClass="div"
          >
            <button
              className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
              id="add-grocery"
              onClick={[MockFunction]}
              type="submit"
            >
              Add Category
            </button>
          </ModalFooter>
        </Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animation": true,
          "aria-labelledby": "contained-modal-title",
          "autoFocus": true,
          "backdrop": true,
          "bsClass": "modal",
          "children": Array [
            <ModalHeader
              bsClass="modal-header"
              closeButton={true}
              closeLabel="Close"
            >
              <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
                id="contained-modal-title"
              >
                Category details
              </ModalTitle>
            </ModalHeader>,
            <ModalBody
              bsClass="modal-body"
              componentClass="div"
              style={
                Object {
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "100%",
                }
              }
            >
              <label
                className="active"
                htmlFor="name"
              >
                Category Title:
              </label>
              <span
                className="alert-danger"
              >
                <br />
                there is error
              </span>
              <input
                className="form-control ca-title"
                defaultValue=""
                id="categoryTitle"
                name="categoryTitle"
                onChange={[MockFunction]}
                required={true}
                type="text"
              />
              <br />
            </ModalBody>,
            <ModalFooter
              bsClass="modal-footer"
              componentClass="div"
            >
              <button
                className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
                id="add-grocery"
                onClick={[MockFunction]}
                type="submit"
              >
                Add Category
              </button>
            </ModalFooter>,
          ],
          "dialogComponentClass": [Function],
          "enforceFocus": true,
          "keyboard": true,
          "manager": ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          },
          "onHide": [MockFunction],
          "renderBackdrop": [Function],
          "restoreFocus": true,
          "show": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-header",
              "children": <ModalTitle
                bsClass="modal-title"
                componentClass="h4"
                id="contained-modal-title"
              >
                Category details
              </ModalTitle>,
              "closeButton": true,
              "closeLabel": "Close",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "modal-title",
                "children": "Category details",
                "componentClass": "h4",
                "id": "contained-modal-title",
              },
              "ref": null,
              "rendered": "Category details",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-body",
              "children": Array [
                <label
                  className="active"
                  htmlFor="name"
                >
                  Category Title:
                </label>,
                <span
                  className="alert-danger"
                >
                  <br />
                  there is error
                </span>,
                <input
                  className="form-control ca-title"
                  defaultValue=""
                  id="categoryTitle"
                  name="categoryTitle"
                  onChange={[MockFunction]}
                  required={true}
                  type="text"
                />,
                <br />,
              ],
              "componentClass": "div",
              "style": Object {
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Category Title:",
                  "className": "active",
                  "htmlFor": "name",
                },
                "ref": null,
                "rendered": "Category Title:",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <br />,
                    "there is error",
                  ],
                  "className": "alert-danger",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  "there is error",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-control ca-title",
                  "defaultValue": "",
                  "id": "categoryTitle",
                  "name": "categoryTitle",
                  "onChange": [MockFunction],
                  "required": true,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-footer",
              "children": <button
                className="waves-effect waves-dark btn right hoverable dark-green btn btn-primary"
                id="add-grocery"
                onClick={[MockFunction]}
                type="submit"
              >
                Add Category
              </button>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Category",
                "className": "waves-effect waves-dark btn right hoverable dark-green btn btn-primary",
                "id": "add-grocery",
                "onClick": [MockFunction],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Add Category",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
